kaldb/src/test/java/com/slack/kaldb/chunk/ChunkManagerTest.java

```diff
@@ -67,24 +75,46 @@ public void setUp() throws InvalidProtocolBufferException {
 
     s3BlobFs = new S3BlobFs();
     s3BlobFs.init(s3Client);
+
+    localZkServer = new TestingServer();
+    localZkServer.start();
+
+    KaldbConfigs.ZookeeperConfig zkConfig =
+        KaldbConfigs.ZookeeperConfig.newBuilder()
+            .setZkConnectString(localZkServer.getConnectString())
+            .setZkPathPrefix(ZK_PATH_PREFIX)
+            .setZkSessionTimeoutMs(15000)
+            .setZkConnectionTimeoutMs(1500)
+            .setSleepBetweenRetriesMs(1000)
+            .build();
+    metadataStoreService = new MetadataStoreService(metricsRegistry, zkConfig);
+    metadataStoreService.startAsync();
+    metadataStoreService.awaitRunning(DEFAULT_START_STOP_DURATION);
   }
 
   @After
-  public void tearDown() throws TimeoutException {
+  public void tearDown() throws TimeoutException, IOException {
     metricsRegistry.close();
+    metadataStoreService.stopAsync();
+    metadataStoreService.awaitTerminated(DEFAULT_START_STOP_DURATION);
     if (chunkManager != null) {
       chunkManager.stopAsync();
       chunkManager.awaitTerminated(DEFAULT_START_STOP_DURATION);
     }
     s3Client.close();
+    if (localZkServer != null) {
```

Comment:
I don't think we need this null check. The `setUp` guarantees that it has been initialized, and none of the other services that are initialized in `@Before` (`s3Client`, `metricsRegistry`, `metadataStoreService`) have null checks. We only are using null checks for `chunkManager` since that's actually initialized in the tests.

Reply:
We employ null checks here since some unit tests kill terminate the ZK server and set this variable to null. It is a convention I adopted for tests, so the teardown code can know this to be the case.

Reply:
For this and the following test however, the `localZkServer` variable is private,  never re-assigned to null, and never passed by reference - making the null check unnecessary.

Reply:
That's true and we don't close ZK in these unit tests for now. However, that's a pattern we used in other tests. Since it's a simple null check, I would like to keep it so I maintain the pattern. Is there any downside of having these null checks?

Reply:
There's a few - it's inconsistent with the other services in the same method (ie, `s3Client`, `metadataStoreService`). It also makes the code less readable - it's suggesting that the client can be set to null when it's impossible with the current code.

Even if the code were refactored at some point in the future such that it could be null - these are tests, so the impact of an NPE would be quick to identify and resolve.

Reply:
Ok. Removed the checks now. I feel more checks don't hurt though.